### PR TITLE
feat: more pitaya metrics

### PR DESCRIPTION
**Pull request by leohahn**
This commit add more metrics to pitaya, particularly latency for client
RPCs and also the number of RPCs in flight currently on the server.

### CODE REVIEW

**Review by victor-carvalho — APPROVED** [review]
LGTM

File: src/pitaya_core/src/metrics.rs [hunk 1]
```diff
@@ -133,16 +133,31 @@ pub fn exponential_buckets(start: f64, factor: f64, count: usize) -> Vec<f64> {
...
         .read()
         .await
         .observe_hist(name, elapsed.as_secs_f64(), labels)
-        .expect("should not fail to observe");
+    {
+        slog::warn!(logger, "observe_hist failed"; "err" => %e);
+    }
+}
+
+pub async fn add_gauge<'a>(
```
Comment (rodopoulos) [inline src/pitaya_core/src/metrics.rs]: Shouldn't you use `register_gauge` instead? We were deprecating the `AddGauge` in NPitaya and using the `Register` prefix.

Comment (leohahn) [reply src/pitaya_core/src/metrics.rs]: This should be `add_to_gauge` in order to make it explicit that we're not registering the metric here, but observing it.

**Comment by rodopoulos** [comment]
LGTM

**Review by rodopoulos — APPROVED** [review]

**Pull request merged**